src/test/java/org/reaktivity/k3po/nukleus/ext/DuplexIT.java

```diff
@@ -318,6 +318,66 @@ public void shouldThrottleInitialOnlyServerSentData() throws Exception
 
     @Test
     @Specification({
+        "server.sent.throttle.stream/client",
+        "server.sent.throttle.stream/server"
+    })
+    public void shouldThrottleClientSentDataPerStream() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should pass because even if throttling is disabled, the receiver window is big enough to receive the message
+    @Test
+    @Specification({
+        "client.sent.throttle.off/client",
+        "client.sent.throttle.off/server"
+    })
+    public void shouldReceiveMessageThrottleOff() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should fail because the receiver window is not big enough and window adjustment is disabled
+    @Test
+    @Specification({
+        "client.sent.throttle.off.negative/client",
+        "client.sent.throttle.off.negative/server"
+    })
+    public void shouldNotReceiveMessageThrottleOff() throws Exception
```

Comment:
If I understand correctly the purpose of this test is to show (a) the client sends data immediately without waiting for a window frame or honoring it (b) the receiver correctly rejects the data because the window has been violated. The rejection is done by sending a Reset frame.
Your test should be testing directly for this rather than just expecting a general test failure. We should build a do this by adding "write aborted" to the client script at the end (after write "Hello, World").
After figuring this out, I looked for a script that is already doing this and found it: server.sent.overflow.

So this test is a duplicate and should be removed. BTW, it would be good to fix the server.sent.overflow/client.rpt script to set nukleus:throttle to "none" instead of "off" (which is now deprecated).

Reply:
ok, removed it and also fixed the throttle option to "none" everywhere it was used.